src/libaktualizr/primary/initializer.cc

```diff
@@ -73,8 +73,15 @@ bool Initializer::initEcuSerials() {
 
 void Initializer::resetEcuSerials() { storage_->clearEcuSerials(); }
 
-// Postcondition: (public, private) is in the storage. It should not be stored until Secondaries are provisioned
-bool Initializer::initPrimaryEcuKeys() { return keys_.generateUptaneKeyPair().size() != 0U; }
+// Postcondition: (public, private) is in the storage. It should not be stored until secondaries are provisioned
+bool Initializer::initPrimaryEcuKeys() {
+  try {
+    return keys_.generateUptaneKeyPair().size() != 0U;
+  } catch (const std::exception& e) {
+    LOG_ERROR << "Initializer error: " << e.what();
+  }
+  return false;
+}
```

Comment:
I'm not sure that's the right place to catch it.
`initPrimaryEcuKeys` gets called only in Initializer ctor. I would definitely prefer if the Initializer ctor actually threw an exception if it failed to do something.

Reply:
Good point, Initializer's methods are worth converting to throwing instead of returning booleans in general.

Reply:
Changed in latest version